src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class BodyTopLevelProperties : TypedRule<Dictionary<string, Operation>>
+    {
+
+        private readonly Regex resourceRefRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private readonly string[] allowedTopLevelProperties = { "name", "type", "id", "location", "properties", "tags", "plan", "sku", "etag",
+                                                                "managedBy", "identity"}; 
+        /// <summary>
+        /// This rule passes if the body parameter contains top level properties only from the allowed set: name, type,
+        /// id, location, properties, tags, plan, sku, etag, managedBy, identity
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid( Dictionary<string, Operation> path, RuleContext context, out object[] formatParameters)
+        {
+            List<string> notAllowedProperties = new List<string>();
+            foreach (var operation in path.Keys)
+            {
+                if ((operation.ToLower().Equals("get") ||
```

Comment:
A suggestion here, we can probably leverage an already implemented function here https://github.com/Azure/autorest/blob/master/src/modeler/AutoRest.Swagger/Validation/TrackedResourceValidation.cs#L159 (we may have to move it to some common utilities namespace)